src/sync/mod.rs

```diff
@@ -1 +1,4 @@
+//! Synchronization primitives.
+
 pub mod list;
```

Comment:
I am not opposed to reexporting structures from internal `crossbeam-rs` crates as part of the public interface, what I mean is rather that this might not be the `List` that we would want to expose. This is a longer discussion that requires its own RFC, but at least my opinion is that the API of this `List` implementation is too low-level. Firstly, it requires the user to provide `Scope` arguments, which the structures currently provided by `crossbeam` don't do. Secondly, the iteration interface has an `Abort` variant. While this is very nice for our purposes, I believe that for an external user who just wants to have a concurrent list, it exposes too much implementation detail and requires said user to understand why exactly iteration can abort, which is a peculiarity of the lock-free implementation.